### PR TITLE
Integrate DAG crawl optimisations from #1777

### PR DESCRIPTION
**Pull request by grumbach**
Integrate DAG crawl changes and optimizations from #1777 including:
- `sn_networking` hard coded version
- parsing of discord names provided by url
- same UTXO delay before retry

Other:
- all DAG crawling back in the same place
- Spends can be sent through a channel for custom processing
- switch from `std::sync`'s `RwLock` to `tokio::sync`'s as the lock is used in tokio co-routines and not in threads

### CODE REVIEW

File: sn_networking/src/version.rs [hunk 1]
```diff
@@ -80,7 +80,7 @@ fn get_truncate_version_str() -> String {
     let version_str = env!("CARGO_PKG_VERSION");
     let parts = version_str.split('.').collect::<Vec<_>>();
     if parts.len() >= 2 {
-        format!("{}.{}", parts[0], parts[1])
+        format!("{}.{}", parts[0], "16")
```
Comment (grumbach) [inline sn_networking/src/version.rs]: Change the version according to #1777

File: sn_client/src/audit/dag_crawling.rs [hunk 2]
```diff
@@ -38,7 +59,7 @@ impl Client {
     pub async fn spend_dag_build_from(
         &self,
         spend_addr: SpendAddress,
-        max_depth: Option<u32>,
+        spend_processing: Option<Sender<SignedSpend>>,
```
Comment (grumbach) [inline sn_client/src/audit/dag_crawling.rs:62]: Optional channel Sender through which we send the Spends collected. 
This is useful for real time external processing of those spends for:
- beta rewards
- following progress
- any other applications (could be used for royalties as well)

File: sn_client/src/audit/dag_crawling.rs [hunk 3]
```diff
@@ -21,6 +24,24 @@ enum InternalGetNetworkSpend {
 }
 
 impl Client {
+    pub async fn new_dag_with_genesis_only(&self) -> WalletResult<SpendDag> {
```
Comment (grumbach) [inline sn_client/src/audit/dag_crawling.rs:27]: Moved this form `sn_auditor` to a `Client` method.

File: sn_networking/src/version.rs (hunk 1 above)
Comment (joshuef) [reply sn_networking/src/version.rs]: This should have been a blocker on that PR. I'll push a commit removing this for `main` integration. (any new network wont need it)

**Review by joshuef — APPROVED** [review]

**Pull request merged**